specification/intelligentdataexperiences/resource-manager/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/preview/2023-03-01-preview/intelligentdataexperiences.json

```diff
@@ -0,0 +1,486 @@
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsResource" },
+            "description": "Parameters supplied to the operation."
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to create AadCustomSecurityAttributesDiagnosticSettings diagnostic setting.",
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsResource" }
+          },
+          "default": {
+            "description": "Error. The request to create diagnostic setting has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          }
+        }
+      },
+      "delete": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_Delete",
+        "description": "Deletes existing diagnostic setting for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. The request to delete named diagnostic setting has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          },
+          "200": { "description": "Successful request to remove a diagnostic setting" },
+          "204": { "description": "Successful request to remove a diagnostic setting" }
+        },
+        "x-ms-examples": {
+          "Deletes the diagnostic setting": { "$ref": "./examples/deleteDiagnosticSetting.json" }
+        }
+      }
+    },
+    "/providers/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/diagnosticSettingsCategories": {
+      "get": {
+        "tags": ["DiagnosticSettingsCategories"],
+        "operationId": "DiagnosticSettingsCategory_List",
+        "description": "Lists the diagnostic settings categories for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. The request to view diagnostic setting categories has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          },
+          "200": {
+            "description": "Successful request to get list of diagnostic setting categories",
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsCategoryResourceCollection" }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic setting categories": { "$ref": "./examples/listDiagnosticSettingsCategories.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ProxyOnlyResource": {
```

Comment:
>ProxyOnlyResource [](http://example.com/codeflow?start=5&length=17)

You should refer common model like ProxyResource etc. from common-types, should not define again locally.

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json)